packages/zoe/src/contracts/constantProduct/README.md

```diff
@@ -0,0 +1,139 @@
+# Constant Product AMM
+
+A constant product automatic market maker based on our Ratio library. It charges
+two kinds of fees: a pool fee remains in the pool to reward the liquidity
+providers and a protocol fee is extracted to fund the economy.
+
+This algorithm uses the x*y=k formula directly, without fees. Briefly, there are
+two kinds of assets, whose values are kept roughly in balance through the
+actions of arbitrageurs. At any time, a trader can come to the pool and offer to
+deposit one of the two assets. They will receive an amount
+of the complementary asset that will maintain the invariant that the product of
+the balances doesn't change. (Except that rounding is done in favor of the
+pool.) The liquidity providers are rewarded by charging a fee. 
+
+The user can specify a maximum amount they want to pay or a minimum amount they
+want to receive. Unlike Uniswap, this approach will charge less than the user
+offered or pay more than they asked for when appropriate. By analogy, if a user
+is willing to pay up to $20 when the price of soda is $3 per bottle, it would
+give 6 bottles and only charge $18. Uniswap doesn't adjust the provided price,
+so it charges $20. This matters whenever the values of the smallest unit of the
+currencies are significantly different, which is common in DeFi. (We refer to
```

Comment:
I didn't understand this sentence. Do you mean, when the two kinds of tokens have very different valuations?